Mathlib/Algebra/Module/LinearMap.lean

```diff
@@ -1014,15 +1014,43 @@ end SMul
 
 section Module
 
-variable [Semiring S] [Module S M₂] [SMulCommClass R₂ S M₂]
+variable [Semiring S] [Module S M₂]
 
-instance module : Module S (M →ₛₗ[σ₁₂] M₂) where
+instance module  [SMulCommClass R₂ S M₂] : Module S (M →ₛₗ[σ₁₂] M₂) where
   add_smul _ _ _ := ext fun _ ↦ add_smul _ _ _
   zero_smul _ := ext fun _ ↦ zero_smul _ _
 
-instance [NoZeroSMulDivisors S M₂] : NoZeroSMulDivisors S (M →ₛₗ[σ₁₂] M₂) :=
+instance [NoZeroSMulDivisors S M₂]  [SMulCommClass R₂ S M₂] :
+    NoZeroSMulDivisors S (M →ₛₗ[σ₁₂] M₂) :=
   coe_injective.noZeroSMulDivisors _ rfl coe_smul
 
+open MulOpposite in
+/-- If `M` is an `(R, S)` bimodule and `M₂` an `S`-module, then `M ⟶ M₂` is also
+  a bimodule-/
+protected def bimodule
+    [Module R M] [Module Sᵐᵒᵖ M] [SMulCommClass R Sᵐᵒᵖ M] [Module Sᵐᵒᵖ M₂] :
+    Module Rᵐᵒᵖ (M →ₗ[Sᵐᵒᵖ] M₂) where
+  smul r l :=
+  { toFun := fun x => l (r.unop • x)
+    map_add' := fun x y => by dsimp; rw [smul_add, map_add]
+    map_smul' := fun s x => by dsimp; rw [smul_comm, map_smul] }
+  one_smul l := LinearMap.ext fun x => show l _ = _ by rw [unop_one, one_smul]
+  mul_smul r₁ r₂ l := LinearMap.ext fun x => show l _ = l _ by rw [unop_mul, mul_smul]
+  smul_zero r := rfl
+  smul_add r l₁ l₂ := LinearMap.ext fun x => show (l₁ + _) _ = _ by
+    rw [LinearMap.add_apply, LinearMap.add_apply]; rfl
+  add_smul r₁ r₂ l := LinearMap.ext fun x => show l _ = l _ + l _ by
+    rw [unop_add, add_smul, map_add]
+  zero_smul l := LinearMap.ext fun x => show l _ = 0 by rw [unop_zero, zero_smul, map_zero]
+
+scoped[Bimodule] attribute [instance] LinearMap.bimodule
```

Comment:
This instance is still a pretty bad idea, such that I think at mimnimum it should be introduced in a more visible and scary way than an `open bimodule` command.

It would probably be better to untangle where this instance is actually needed, then remove it entirely.

Reply:
I removed this instance and moved it to `TensorHomAdjunction.lean` as a local instance in [12eeac9](https://github.com/leanprover-community/mathlib4/pull/8495/commits/12eeac90d291b69a9b2fffd4f8d2cc1052343910), is this helpful?